### PR TITLE
Pre-judgment database name is null.

### PR DESCRIPTION
**Pull request by zjcnb**

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java [hunk 1]
```diff
@@ -98,6 +98,9 @@ public InsertStatementContext(final Map<String, ShardingSphereDatabase> database
     
     private ShardingSphereSchema getSchema(final Map<String, ShardingSphereDatabase> databases, final String defaultDatabaseName) {
         String databaseName = tablesContext.getDatabaseName().orElse(defaultDatabaseName);
+        if (null == databaseName) {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java:101]: @zjcnb In what scenario will databaseName be empty?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java (hunk 1 above)
Comment (zjcnb) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java:101]: Auto send the `select verison()` command after connect Proxy, the`databaseName` are empty At this time.

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/parse/PostgreSQLComParseExecutorTest.java [hunk 2]
```diff
@@ -121,6 +125,13 @@ public void assertExecuteWithParameterizedSQL() {
         assertThat(actualPreparedStatement.getParameterTypes(), is(Arrays.asList(PostgreSQLColumnType.POSTGRESQL_TYPE_INT4, PostgreSQLColumnType.POSTGRESQL_TYPE_UNSPECIFIED)));
     }
     
+    @SneakyThrows
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/parse/PostgreSQLComParseExecutorTest.java:128]: Please throw concrete exception

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java [hunk 3]
```diff
@@ -128,15 +128,17 @@ private Map<Integer, SelectStatementContext> createSubqueryContexts(final Map<St
...
     
     private Map<String, ShardingSphereSchema> getSchemas(final Map<String, ShardingSphereDatabase> databases, final String databaseName) {
-        ShardingSphereDatabase database = databases.get(databaseName);
-        if (null == database) {
+        if (null == databaseName) {
             if (tablesContext.getTables().isEmpty()) {
                 return Collections.emptyMap();
             } else {
-                throw new DatabaseNotExistedException(databaseName);
+                throw new DatabaseNotExistedException(null);
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java:135]: null parameter maybe not good, how about create a new exception?
maybe `NoDatabaseException` is a good name.

Comment (zjcnb) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java:135]: Yeah, i agree. I will adjustment it

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/parse/PostgreSQLComParseExecutorTest.java (hunk 2 above)
Comment (zjcnb) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/src/test/java/org/apache/shardingsphere/proxy/frontend/postgresql/command/query/extended/parse/PostgreSQLComParseExecutorTest.java:128]: Sure

**Review by strongduanmu — APPROVED** [review]
LGTM.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**